src/components/PickerKeyboard.js

```diff
@@ -44,6 +44,11 @@ class PickerKeyboard extends Component {
   onCancelPress() {
     this.setVisible(false);
 
+    // When press cancel will set current value back to picker state
+    this.setState({
+      value: this.props.value
+    });
+
```

Comment:
So what if someone has already changed the value, then opens the picker again, and presses cancel. I don't think the expected behaviour would be to reset it to the the props value?